src/app/core/utils/translate/icm-translate-loader.ts

```diff
@@ -1,47 +1,40 @@
-import { Injectable } from '@angular/core';
-import { TransferState } from '@angular/platform-browser';
-import { Actions, ROOT_EFFECTS_INIT, ofType } from '@ngrx/effects';
-import { Store, select } from '@ngrx/store';
+import { isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { TransferState, makeStateKey } from '@angular/platform-browser';
 import { TranslateLoader } from '@ngx-translate/core';
-import { ReplaySubject, combineLatest, defer, from, of } from 'rxjs';
-import { catchError, first, map, switchMap, take, tap } from 'rxjs/operators';
+import { combineLatest, defer, from, iif, of } from 'rxjs';
+import { catchError, map, tap } from 'rxjs/operators';
 
-import { NGRX_STATE_SK, STATE_ACTION_TYPE } from 'ish-core/configurations/ngrx-state-transfer';
-import { getServerTranslations, loadServerTranslations } from 'ish-core/store/core/configuration';
-import { whenTruthy } from 'ish-core/utils/operators';
+import { LocalizationsService } from 'ish-core/services/localizations/localizations.service';
+
+import { Translations } from './translations.type';
 
 @Injectable()
 export class ICMTranslateLoader implements TranslateLoader {
-  private initialized$ = new ReplaySubject(1);
-
-  constructor(private store: Store, actions: Actions, transferState: TransferState) {
-    const actionType = transferState.hasKey(NGRX_STATE_SK) ? STATE_ACTION_TYPE : ROOT_EFFECTS_INIT;
-    actions.pipe(ofType(actionType), first()).subscribe(() => {
-      this.initialized$.next(true);
-    });
-  }
+  constructor(
+    private transferState: TransferState,
+    @Inject(PLATFORM_ID) private platformId: string,
+    private localizations: LocalizationsService
+  ) {}
 
   getTranslation(lang: string) {
+    const SSR_TRANSLATIONS = makeStateKey<Translations>('ssrTranslations-' + lang);
+
     const local$ = defer(() => from(import(`../../../../assets/i18n/${lang}.json`)).pipe(catchError(() => of({}))));
-    const server$ = this.store.pipe(
-      select(getServerTranslations(lang)),
-      tap(translations => {
-        if (!translations) {
-          this.store.dispatch(loadServerTranslations({ lang }));
-        }
-      }),
-      whenTruthy(),
-      take(1)
-    );
-    return this.initialized$.pipe(
-      switchMap(() =>
-        combineLatest([local$, server$]).pipe(
-          map(([localTranslations, serverTranslations]) => ({
-            ...localTranslations,
-            ...serverTranslations,
-          }))
-        )
+    const server$ = iif(
+      () => isPlatformBrowser(this.platformId) && this.transferState.hasKey(SSR_TRANSLATIONS),
+      of(this.transferState.get(SSR_TRANSLATIONS, {})),
+      this.localizations.getServerTranslations(lang).pipe(
```

Comment:
Now that the loader uses the localizations service directly, the ´loadServerTranslations´ actions have become obsolete. We should remove them and their accompanying effect & reducer logic.

Reply:
👍